boot_services/src/boot_services.rs

```diff
@@ -1487,4 +1487,17 @@ mod test {
         let status = boot_services.free_pages(0x100000, 10);
         assert!(matches!(status, Ok(())));
     }
+
+    #[test]
+    fn test_free_pool() {
+        let boot_services = boot_services!(free_pool = efi_free_pool);
+
+        extern "efiapi" fn efi_free_pool(buffer: *mut c_void) -> efi::Status {
+            assert_eq!(buffer, 0xffff0000 as *mut u8 as *mut c_void);
+            efi::Status::SUCCESS
+        }
+
+        let status = boot_services.free_pool(0xffff0000 as *mut u8);
+        assert_eq!(status, Ok(()));

```

Comment:
you could also add a negative test, by having efI_free_pool return an error status - you could either do this as a nother step in this test or add another test.

Reply:
I add a negative test in this test.